mesonbuild/build.py

```diff
@@ -255,7 +255,8 @@ def __init__(self, environment: environment.Environment):
         self.data: T.List[Data] = []
         self.symlinks: T.List[SymlinkData] = []
         self.static_linker: PerMachine[StaticLinker] = PerMachine(None, None)
-        self.subprojects = {}
+        # TODO: Shouldn't this be `dict[SubProject, str]`? That would mean some changes to the interpreter
+        self.subprojects: T.Dict[str, str] = {}  # Maps pretty subproject name to it's version
```

Comment:
The changes to SubProject don't look significant.

https://github.com/mesonbuild/meson/compare/master...eli-schwartz:meson:for-dcbaker/misc-typing